rotkehlchen/tasks/assets.py

```diff
@@ -196,10 +202,88 @@ def update_owned_assets(user_db: DBHandler) -> None:
         )
 
 
+def _aave_tokens_are_correct(
+        token_to_underlying: dict['ChecksumEvmAddress', 'ChecksumEvmAddress'],
+        chain_id: ChainID,
+) -> list['ChecksumEvmAddress']:
+    """Check if aave tokens are stored in the globaldb.
+    We consider a token as stored when it has the aave v3 protocol set. This means that
+    it was not added by decoders but by the logic of update_aave_v3_underlying_assets.
+    """
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        cursor.execute(
+            'SELECT address, protocol FROM evm_tokens '
+            f"WHERE address IN ({','.join('?' * len(token_to_underlying))}) "
+            'AND chain=?',
+            tuple(token_to_underlying.keys()) + (chain_id.serialize_for_db(),),
+        )
+
+        missing_tokens = set(token_to_underlying.keys())
+        for row in cursor:
+            if row[1] == CPT_AAVE_V3:
+                missing_tokens.remove(row[0])
+
+    output = list(missing_tokens)  # we return a sorted list to avoid issues with cassettes
```

Comment:
Doing more work in production just to make VCR easier is not something we should do.

How about abstracting this in a way that can be mocked in the tests and adds the sorting only for tests?